### PR TITLE
Relationship not getting store fix

### PR DESCRIPTION
**Pull request by ansjin**

### CODE REVIEW

File: app/dbConnection/index.js [hunk 1]
```diff
@@ -239,7 +247,11 @@ exports.writeEvents = function (eventEntityJSON) {
...
       return artist.getEntity().then((artistEntity)=> {
         if(!artistEntity) {
-          return entities.create()
+          return entities.create({
+'name': entityName,
+'artist_type': 'Composer'
+
+})
             .then(newEntity => {
               return artist.setEntity(newEntity).then(() => newEntity);
```
Comment (Sandr0x00) [inline app/dbConnection/index.js]: Maybe we should NOT set that here? @ansjin what do you think?
The artist.setEntity(...) stuff.

File: app/dbConnection/index.js [hunk 2]
```diff
@@ -232,15 +328,17 @@ exports.writeEvents = function (eventEntityJSON) {
     let artists = context.models.artists;
     let entities = context.models.entities;
     let eventJSON = eventEntityJSON.content;
-    return artists.findOrCreate({ // TODO: change to findOne
+    return artists.find({
```
Comment (Sandr0x00) [inline app/dbConnection/index.js]: Reason for null-artists, Before merging, please check this case as well!

File: app/dbConnection/index.js [hunk 3]
```diff
@@ -72,7 +72,7 @@ exports.getPromisingWebsites = function () {
       console.log('Query the db');
       // TODO: remove the limit
       context.sequelize.query(
-        'SELECT * from (SELECT blob_url, COUNT(*) as count FROM (select * from websites limit 1000) w LEFT JOIN contains c ON w.id = c."websiteId" GROUP BY blob_url ORDER BY count DESC) a where a.count > 50;',
+        'SELECT * from (SELECT blob_url, COUNT(*) as count FROM (select * from websites limit 100000 ) w LEFT JOIN contains c ON w.id = c."websiteId" GROUP BY blob_url ORDER BY count DESC) a where a.count > 5;',
```
Comment (Henni) [inline app/dbConnection/index.js]: * [x] remove limit
* [ ] make threshold configurable and pass it into the query

**Review by Henni — CHANGES_REQUESTED** [review]

File: app/dbConnection/index.js [hunk 4]
```diff
@@ -232,15 +328,17 @@ exports.writeEvents = function (eventEntityJSON) {
...
       }
-    }).spread((artist, created) => {
+    }).then(artist => {
       return artist.getEntity().then((artistEntity)=> {
         if(!artistEntity) {
-          return entities.create()
-            .then(newEntity => {
+          return entities.create({
+              'name': entityName,
+              'artist_type': 'Composer'
```
Comment (Sandr0x00) [inline app/dbConnection/index.js]: Entity has no artist type, but artist has.

File: app/dbConnection/index.js [hunk 5]
```diff
@@ -220,6 +235,72 @@ exports.writeRelationships = function (relationJSON) {
...
+      });
+    } else if (foundInstruments.length > 1) {
+      // more instruments found, better not link
+    } else {
+      // we found nothing, no linking at all
+    }
+  });
+}
+
 exports.writeEvents = function (eventEntityJSON) {
```
Comment (Sandr0x00) [inline app/dbConnection/index.js]: @ansjin I do not understand this method, can you check that again please?
Problems:
- The stuff i wrote in other comments
- In the last then, we collect all promises, but then ignore them?
- We should create a event, even if we have no corresponding artist or entity.
- We should not create artists!
- We should not create entities? Or should we? Makes no sense in my Opinion.

I think the best way would be to create `relationshipEntities`. Maybe.

We should check if `entityName` is `undefined` or `null`, before creating null-entries ;)
Where do we get the `entityName`? Does the algorithm already know the entity?

**Review by Sandr0x00 — CHANGES_REQUESTED** [review]

**Review by Sandr0x00 — APPROVED** [review]

**Comment by ansjin** [comment]
no merging until we test this!

**Comment by kordianbruck** [comment]
@anshul0708 @Sandr00 can we get this merged please, so we have it for our run?

**Comment by ansjin** [comment]
@kordianbruck  you tagged here the other anshul :)

**Comment by ansjin** [comment]
@Henni Please approve this!

**Review by Henni — APPROVED** [review]

**Comment by kordianbruck** [comment]
@ansjin happens to me every time :cry:

**Pull request merged**